rasa/core/brokers/pika.py

```diff
@@ -399,27 +426,170 @@ def _on_channel_open(self, channel: "Channel") -> None:
 
         self.channel = channel
 
-        while self._unpublished_messages:
-            # Send unpublished messages
-            message = self._unpublished_messages.popleft()
-            self._publish(message)
+    def _publish(self, message: Message) -> None:
+        (body, headers) = message
+
+        self.channel.basic_publish(
+            exchange=RABBITMQ_EXCHANGE,
+            routing_key="",
+            body=body.encode(DEFAULT_ENCODING),
+            properties=self._get_message_properties(headers),
+        )
+
+    def process_messages(self) -> None:
+        """Start to process messages. This process is indefinite thus it should
+        be started in a separate thread or process.
+
+        This method also automatically establishes a connection and wait for
+        the channel to get ready to process messages.
+        """
+        if not self.is_connected:
+            self._connect()
+
+        if not self.is_ready():
+            logger.warning("RabbitMQ channel cannot be opened.")
+            return
+
+        try:
+            while True:
+                message = self.get_message()
```

Comment:
will this block if there's no message in the queue?

Reply:
yep!
(this part was actually copied from the previous logic)